js/ui/iconGridLayout.js

```diff
@@ -379,6 +379,77 @@ const IconGridLayout = new Lang.Class({
                 Lang.bind(this, this._enumerateDirectoryFiles));
     },
 
+    _createFolderFile: function(name) {
+        let keyFile = new GLib.KeyFile();
+
+        keyFile.set_value(GLib.KEY_FILE_DESKTOP_GROUP,
+                          GLib.KEY_FILE_DESKTOP_KEY_NAME, name);
+
+        keyFile.set_value(GLib.KEY_FILE_DESKTOP_GROUP,
+                          GLib.KEY_FILE_DESKTOP_KEY_TYPE, GLib.KEY_FILE_DESKTOP_TYPE_DIRECTORY);
+
+        let dir = Gio.File.new_for_path(GLib.get_user_data_dir() + '/desktop-directories');
+        if (!dir.query_exists(null))
+            try {
+                dir.make_directory_with_parents(null);
+            } catch (err) {
+                log('Error creating %s: %s'.format(dir.get_path(), err.message));
+                return null;
+            }
+
+        let enumerator = null;
+        try {
+            enumerator = dir.enumerate_children('standard::name', Gio.FileQueryInfoFlags.NONE, null);
+        } catch (err) {
+            log('Error trying to traverse %s: %s'.format(dir.get_path(), err.message));
+            return null;
+        }
+
+        let prefix = 'eos-folder-user-';
+        let suffix = '.directory';
+        let re = new RegExp(prefix + '([0-9]+)' + suffix);
+        let n = -1;
+
+        try {
+            for (let f = enumerator.next_file(null); f != null; f = enumerator.next_file(null)) {
+                let result = re.exec(f.get_name());
+                if (result != null) {
+                    let new_n = result[1];
+                    n = Math.max(n, parseInt(new_n));
+                }
+            }
+        } catch (err) {
+            log('Error traversing %s: %s'.format(dir.get_path(), err.message));
+            return null;
+        }
+
+        try {
+            enumerator.close(null);
+        } catch (err) {
+            log('Error closing file enumerator for %s: %s'.format(dir.get_path(), err.message));
```

Comment:
Use `logError()` instead -- but I don't think this error should be considered fatal